vllm/config.py

```diff
@@ -3278,6 +3285,15 @@ def _extract_backend_options(self):
             self.disable_additional_properties = True
 
 
+# For backward compatibility, should remove in v0.10.0
+# given that vllm.config is considered public
+@deprecated(
+    "DecodingConfig is deprecated. Uses 'from vllm.config import StructuredOutputConfig' instead"  # noqa: E501
```

Comment:
Can you update the deprecation warning to say:

- What's deprecated
- When it's going away
- What the user should do instead

That way the user gets all the information they need to fix their code in the warning that gets logged